src/main/java/com/github/princesslana/eriscasper/data/resource/GuildMemberWithGuildId.java

```diff
@@ -2,15 +2,36 @@
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.databind.JsonNode;
+import com.github.princesslana.eriscasper.data.Data;
 import com.github.princesslana.eriscasper.data.DataException;
+import com.github.princesslana.eriscasper.data.Snowflake;
+
+public class GuildMemberWithGuildId {
+
+  private Snowflake guildId;
+  private GuildMember member;
+
+  private GuildMemberWithGuildId() {}
 
-public class GuildMemberWithGuildId extends WithGuildId<GuildMember> {
   @JsonCreator
   public GuildMemberWithGuildId(JsonNode json) throws DataException {
-    super(GuildMember.class, json);
+    guildId = Snowflake.of(json.get("guild_id").asText());
+    member = Data.fromJson(json, GuildMember.class);
   }
 
+  public Snowflake getGuildId() {
+    return guildId;
+  }
+
+  @SuppressWarnings("unused")
   public GuildMember getGuildMember() {
-    return getValue();
+    return member;
+  }
+
+  public static GuildMemberWithGuildId from(Snowflake guildId, GuildMember member) {
```

Comment:
I would prefer `fromJson` to be a static method, and have it call a constructor that set these fields.